rust/headless-client/src/windows.rs

```diff
@@ -186,22 +186,72 @@ impl IpcServer {
     /// This is async on Linux
     #[allow(clippy::unused_async)]
     pub(crate) async fn new() -> Result<Self> {
+        Self::new_with_path(pipe_path())
+    }
+
+    /// Uses a test path instead of what prod uses
+    ///
+    /// The test path doesn't need admin powers and won't conflict with the prod
+    /// IPC service on a dev machine.
+    ///
+    /// This is async on Linux
+    #[allow(clippy::unused_async)]
+    #[cfg(test)]
+    pub(crate) async fn new_for_test() -> Result<Self> {
+        let pipe_path = named_pipe_path(&format!("{BUNDLE_ID}_test.ipc_service"));
+        Self::new_with_path(pipe_path)
+    }
+
+    pub(crate) fn new_with_path(pipe_path: String) -> Result<Self> {
         setup_before_connlib()?;
-        Ok(Self {})
+        Ok(Self { pipe_path })
     }
 
+    // `&mut self` needed to match the Linux signature
     pub(crate) async fn next_client(&mut self) -> Result<IpcStream> {
-        let server = create_pipe_server()?;
+        // Fixes #5143. In the IPC service, if we close the pipe and immediately re-open
+        // it, Tokio may not get a chance to clean up the pipe. Yielding seems to fix
+        // this in tests, but `yield_now` doesn't make any such guarantees, so
+        // we also do a loop.
+        tokio::task::yield_now().await;
```

Comment:
if we are looping we no longer need this, right?

Reply:
This does the fix by itself, the loop is just defense-in-depth since yielding isn't guaranteed to do anything specific.

With `yield_now` the loop doesn't trigger:
<img width="564" alt="image" src="https://github.com/firezone/firezone/assets/13400041/92f6b56f-a248-40c4-b2f0-b6ba8b622cf1">

With `yield_now` commented out, usually one loop will save it:
<img width="568" alt="image" src="https://github.com/firezone/firezone/assets/13400041/82579b55-3fc5-44c7-9997-9dece1a858cf">

Reply:
`yield_now()` so that another task can take over and fixing the bug feels a bit like a hack, I'd rather let it loop once makes what's happening a bit more obvious